scripts/git_changes_utils.py

```
@@ -0,0 +1,304 @@
+            if not err:
+                if remote_url.endswith(b'oppia/oppia.git\n'):
+                    remote_num += 1
+                    remote_name = remote
+            else:
+                raise ValueError(err)
+    else:
+        raise ValueError(err)
+
+    if not remote_num:
+        raise Exception(
+            'Error: Please set the git upstream.\n'
+            'To do that follow these steps:\n'
+            '1. Run the command \'git remote -v\'\n'
+            '2a. If upstream is listed in the command output, then run the '
+            'command \'git remote set-url upstream '
+            'https://github.com/oppia/oppia.git\'\n'
+            '2b. If upstream is not listed in the command output, then run the '
+            'command \'git remote add upstream '
+            'https://github.com/oppia/oppia.git\'\n'
+        )
+
+    if remote_num > 1:
+        print(
+            'Warning: Please keep only one remote branch for oppia:develop.\n'
+            'To do that follow these steps:\n'
+            '1. Run the command \'git remote -v\'\n'
+            '2. If there are multiple remotes listed for upstream, then run '
+            'the command, \'git remote remove <remote_name>\' for all remotes '
+            'that are not the main oppia repository.\n'
+        )
+        return None
+    return remote_name
+
+
+def git_diff_name_status(
+    left: Optional[str] = None, right: Optional[str] = None,
+    diff_filter: Optional[str] = None
+) -> List[FileDiff]:
+    """Compare two branches/commits with git.
+
+    Parameter:
+        left: str. The name of the lefthand branch.
+        right: str. The name of the righthand branch.
+        diff_filter: str. Arguments given to --diff-filter (ACMRTD...).
+
+    Returns:
+        list. List of FileDiffs (tuple with name/status).
+
+    Raises:
+        ValueError. Raise ValueError if git command fails.
+    """
+    git_cmd = ['git', 'diff', '--name-status']
+    if diff_filter:
```

You're now using `None` instead of `''` as the representation for an unspecified argument. I agree with this change, but now `''` shouldn't be treated as an unspecified argument (it should probably throw an error). We don't want people passing `''` as a way to mean that no `diff_filter` is specified.

Alternatively, if you do want to support `''` as a kind of null value, then please document this so that future developers know to preserve that behavior

---

Done.